### PR TITLE
[mod_admin_stats] - add links to users and articles

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #13386 .

### Summary of Changes

- added link to users item to com_users
- added link to articles item to com_content
- ~~added link to OS item to com_admin&view=sysinfo#site~~
- ~~added link to PHP item to com_admin&view=sysinfo#phpinfo~~
- ~~added link to DB item to com_config#page-server~~
- ~~added link to Time item to com_config#page-server~~
- ~~added link to Caching item to com_config#page-system~~
- ~~added link to Gzip item to com_config#page-system~~

### Testing Instructions
0. code review
1. apply the patch
2. Extensions -> Modules -> Statistics
3. Enable `Site Information`
4. the `Users` & `Articles` items are now linked to their components

linkable

### Additional Info
integrated with weblinks see #13389 and https://github.com/joomla-extensions/weblinks/pull/288

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 0d803138cd2dff0ab665356f2e5bc4a39665be12<br /><br />Works as expected; thanks for Enhancement.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by ralain** [comment]
I have tested this item :white_check_mark: successfully on 0d803138cd2dff0ab665356f2e5bc4a39665be12<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by zero-24** [comment]
Hmm i don't want to sound bad but what is the reason for that link?

For me the admin UI is now even more complicated as we have 9 entrys (+ weblinks) but just 2/3 with a link? And i don't see the reason for just the third place we link to com_content

**Comment by chrisdavenport** [comment]
@alikon I'm just wondering if it might be better to style the count in the same way we do on, for example, the category screens.  Then have the count be a clickable link, rather than the text.  That might be a little more consistent with those other screens.  What do you think?

**Comment by alikon** [comment]
@zero-24 one click less from control panel to Users & Articles
for the others module items i was thinking to make those linkable as well, for example :

- Caching -> link to com_config and open the system tab 
- databaase -> link to com_config and open the server tab 
- etc
but i'm unable to do the js code to link the specific tab any help on this ?

@chrisdavenport  done 
chris

**Comment by rdeutz** [comment]
I am setting this to needs review, would be good to hear something from the ui/ux team on this @cpfeifer <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by dgt41** [comment]
> but i'm unable to do the js code to link the specific tab any help on this ?

I can help here, let me prepare something

**Comment by alikon** [comment]
thanks @dgt41, i was hoping for some help by js expert like you

**Comment by dgt41** [comment]
@alikon check https://github.com/alikon/joomla-cms/pull/26

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on 9c53d4b38b99622909c8565021734fd90c37e6bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by cpfeifer** [comment]
@rdeutz I think this is a great improvement. Anything we can do to add useful interactions to the dashboard is welcomed in my opinion.

**Comment by mbabker** [comment]
Personal opinion, I think this is just adding links for the sake of adding links.  It's a statistics module intended to give a rollup of various data points (and finally pluggable in 3.7), I don't see it as an element that should be providing site navigation.

The linked counts in the category views have a useful purpose, they provide quick filters to a given category and publication state in the component's item list view.  What purpose do these links serve?

**Comment by zero-24** [comment]
About short links we already have them with one click on the left sidebar for users and articles ;)

Is it not more confusing to have 100 options on one screen that do the same?

**Comment by franz-wohlkoenig** [comment]
@zero-24 You're right about confusing, but there's one Link for user as Example, other Link is behind a Menu. Its about User, to choose which way benefit her. Personally i like Shortcuts, another way to improve Joomla!.

**Comment by cpfeifer** [comment]
I like it because it provides a fast and easy way for users to see where the stats are coming from. If it says "20 users" for example, and that doesn't seem right to the admin they can go directly to the user manager and see what is happening.

Plus, some of the dashboard stat module concepts I'm currently working on include these links so I think this is a good step towards the future. I'm still working that out, and this change may not have a huge impact in 3.7, but there will be more of this kind of thing coming.

**Comment by cpfeifer** [comment]
I have tested this item :white_check_mark: successfully on 9c53d4b38b99622909c8565021734fd90c37e6bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by dgt41** [comment]
@alikon can you merge my PR?

**Comment by mbabker** [comment]
Maybe with those new concepts it might make sense, but in the current module's build, this honestly seriously feels like it's shoving more links in place for the sake of shoving more links in.  Especially with the discussion about linking the server metric items to a specific tab on a specific page.  I do not see how this feature is more beneficial than not.

Yes, I get it might be a convenient shortcut, but when you stop focusing on this module without any additional context and look at its general use, it is displayed on a page where all of these resources are at most a couple of clicks away.  It feels like overkill to me; do we really need 3 or possibly more links to the Article Manager on the admin control panel on one page?

**Comment by andrepereiradasilva** [comment]
IMHO it would make more sense if a link was in the number itself ...

**Comment by alikon** [comment]
@dgt41 see https://github.com/alikon/joomla-cms/pull/26#issuecomment-269803190

**Comment by cpfeifer** [comment]
The link is in the number and the change is very small. Design-wise it's basically adding a color highlight to page element which already exists while providing a shortcut. I don't see any downsides to this change.

File: administrator/modules/mod_stats_admin/helper.php [hunk 1]
```diff
@@ -112,6 +112,7 @@ public static function getStats(&$params)
 				$rows[$i]->title = JText::_('MOD_STATS_USERS');
 				$rows[$i]->icon  = 'users';
 				$rows[$i]->data  = $users;
+				$rows[$i]->link  = JUri::root() . 'administrator/index.php?option=com_users';
```
Comment (andrepereiradasilva) [inline administrator/modules/mod_stats_admin/helper.php]: can we not hardcode administrator here ?
```php
JRoute::_('index.php?option=com_users');
```

File: administrator/modules/mod_stats_admin/helper.php [hunk 2]
```diff
@@ -121,6 +122,7 @@ public static function getStats(&$params)
 				$rows[$i]->title = JText::_('MOD_STATS_ARTICLES');
 				$rows[$i]->icon  = 'file';
 				$rows[$i]->data  = $items;
+				$rows[$i]->link  = JUri::root() . 'administrator/index.php?option=com_content&view=articles&filter[published]=1';
```
Comment (andrepereiradasilva) [inline administrator/modules/mod_stats_admin/helper.php]: can we not hardcode administrator here ?
```php
JRoute::_('index.php?option=com_content&view=articles&filter[published]=1');
```

**Comment by mbabker** [comment]
That PR to add the JavaScript to link to specific tabs honestly makes me like this less.  But hey, if everyone's OK with a UI where almost every element is navigable in some form and causes actions to be duplicated on the same screen then who am I to stop it?

Linking the items is one thing.  Hiding the link in a badge that doesn't even look like a clickable element just makes it worse.

**Comment by mbabker** [comment]
And just for all fairness, I think it's equally bad with WordPress' default dashboard.

screen shot 2016-12-30 at 12 31 54 pm

The shortcuts there are to items that are 250 pixels to the left (from the far right edge of posts to the far left edge of the icons) and 330 pixels above (from bottom edge of posts to the top of its icon).  Seriously it just feels like link shoving to me.  It'd be another argument altogether if it were the only navigation element to that section on the page or added some value (a filter or a quick add/edit action), but it's not either here nor with WordPress.  And here we're adding JavaScript to make a link take someone to a specific tab on a specific page; are we going to add anchors to pages to drop someone on a specific line or add more JavaScript to highlight the specific section of the page too?

File: administrator/modules/mod_stats_admin/helper.php (hunk 1 above)
Comment (alikon) [reply administrator/modules/mod_stats_admin/helper.php]: done
thanks

File: administrator/modules/mod_stats_admin/helper.php (hunk 2 above)
Comment (alikon) [reply administrator/modules/mod_stats_admin/helper.php]: done
thanks

**Comment by alikon** [comment]
thanks to @dgt41 now all items are linkable and open the correct tab
linkable

**Comment by frankmayer** [comment]
A lot of times I miss "interconnection" and easier moving around between things in Joomla. I think this PR is a good step forward into that direction.

However, I think @mbabker is right, in that it's not that clear to the user, that the link is "hidden" in a badge.

How about using buttons instead of labels? Like this (like the other buttons that also make it clear for the user that it is a something clickable).

nimbus-image-1483474033126

This is only using `btn btn-info` instead of `badge badge-info`.

Or maybe just put links on the labels, for example OS, PHP etc... like in the "RECENTLY ADDED ARTICLES" above.

**Comment by mbabker** [comment]
That's even worse stylistically.  Honestly, I think I could live with it if just the users and articles items were linked, but I seriously think all this extra linking for the server metrics is taking it too far.  What is really being gained by direct linking the time stamp to the server tab on the global config here, especially as the global timezone is the configuration item and the time is NOT a manageable item in a website (unless you know of a PHP app that lets you change the physical server's time)?  This is where it feels like we're link stuffing for the sake of linking all the things and why in general I'm adamantly against this PR.  Someone took an idea and went too far with it.

**Comment by alikon** [comment]
i like the `btn btn-info`
let's see what the maintainers decide

**Comment by jeckodevelopment** [comment]
Well, personal opinion on that:
- In favour of linking "Articles", "Users", "Weblinks".
- Against all the linked configuration values. 

Have all these links to me is almost meaningless.

**Comment by frankmayer** [comment]
@mbabker stylistically, I like the `btn` version better than the `badge`, though I admit, the `btn` ought to be a bit narrower to not be that bulky, which I guess is the stylistic concern you have. But keep in mid in was only a quick demo show an example, here.

**Comment by mbabker** [comment]
Honestly I think the badges work fine but they really aren't intended to be used as navigable elements (same issue exists elsewhere in the backend where badges are linked).  Using buttons would definitely fix that, but I'm going to assume (because I don't know what exists in Joomla Bootstrap anymore) there's going to be yet another button variant added to the CSS for this just to deal with the margin/padding.

**Comment by brianteeman** [comment]
@mbabker as release lead for 3.8 can we have a decision on this please so that it is either accepted or rejected. No point in asking for merge conflicts to be resolved if it is either going to sit here forever or rejected

**Comment by mbabker** [comment]
Link support in general, I can live with.  Linking to managers for those items, feels like link stuffing but if it helps users then OK I guess.  Linking every item to either a specific tab or field, that's the part I don't like at all.  Keep scope to linking articles and users and I'm OK with this.

**Comment by brianteeman** [comment]
@alikon 
Can you look at resolving the conflicts so that this can be tested please and the comment from @mbabker

**Comment by alikon** [comment]
conflict resolved 
link only on articles and users

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on b2537d3e574bd823c455bc07a2653768cdfc0819<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by Quy** [comment]
This module is no longer installed in v3.8. Do you know why?

With the styling, these links are very prominent. I prefer them to be subtle like links in Recently Added Articles.

**Comment by ghost** [comment]
@alikon any Update on this PR?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by alikon** [comment]
What update @franz-wohlkoenig  ?

**Comment by ghost** [comment]
Comment by @Quy above:

> This module is no longer installed in v3.8. Do you know why?
>
> With the styling, these links are very prominent. I prefer them to be subtle like links in Recently Added Articles.

**Comment by alikon** [comment]
the module is still there https://github.com/joomla/joomla-cms/blob/staging/installation/sql/mysql/joomla.sql#L564

**Comment by Sieger66** [comment]
I have tested this item :white_check_mark: successfully on b2537d3e574bd823c455bc07a2653768cdfc0819<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13396">issues.joomla.org/tracker/joomla-cms/13396</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**